### PR TITLE
[dapp-kit] export useResolveSuiNSName and show SuiNS name in account dropdown

### PR DESCRIPTION
**Pull request by hayes-mysten**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: sdk/dapp-kit/src/components/AccountDropdownMenu.tsx [hunk 1]
```diff
@@ -67,3 +64,25 @@ export function AccountDropdownMenu({ currentAccount }: AccountDropdownMenuProps
...
 }
+
+export function AccountDropdownMenuItem({
+	account,
+	active,
+}: {
+	account: WalletAccount;
+	active?: boolean;
+}) {
+	const { mutate: switchAccount } = useSwitchAccount();
```
Comment (hayes-mysten) [inline sdk/dapp-kit/src/components/AccountDropdownMenu.tsx:75]: @williamrobertson13 should I move this out into the parent and pass a callback prop? Not sure if there is a lot of overhead

File: sdk/dapp-kit/src/components/AccountDropdownMenu.tsx [hunk 2]
```diff
@@ -67,3 +64,25 @@ export function AccountDropdownMenu({ currentAccount }: AccountDropdownMenuProps
...
+	const { data: domain } = useResolveSuiNSName(account.label ? null : account.address);
+
+	return (
+		<DropdownMenu.Item
+			key={account.address}
```
Comment (williamrobertson13) [inline sdk/dapp-kit/src/components/AccountDropdownMenu.tsx:80]: no need for a key here

File: sdk/dapp-kit/src/components/AccountDropdownMenu.tsx (hunk 1 above)
Comment (williamrobertson13) [reply sdk/dapp-kit/src/components/AccountDropdownMenu.tsx:75]: there's not really any overhead, this should be fine!

File: sdk/dapp-kit/src/hooks/useResolveSuiNSNames.ts [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
+import type { ResolvedNameServiceNames } from '@mysten/sui.js/client';
+import type { UseQueryOptions } from '@tanstack/react-query';
+
 import { useSuiClientQuery } from './useSuiClientQuery.js';
 
-export function useResolveSuiNSName(address?: string | null) {
+export function useResolveSuiNSName(
+	address?: string | null,
+	options?: Omit<
+		UseQueryOptions<ResolvedNameServiceNames, Error, ResolvedNameServiceNames, unknown[]>,
```
Comment (williamrobertson13) [inline sdk/dapp-kit/src/hooks/useResolveSuiNSNames.ts:12]: should we exclude `enabled` from here?

**Review by williamrobertson13 — APPROVED** [review]
lgtm!

File: sdk/dapp-kit/src/hooks/useResolveSuiNSNames.ts (hunk 3 above)
Comment (hayes-mysten) [reply sdk/dapp-kit/src/hooks/useResolveSuiNSNames.ts:12]: No, I still want to be be able to disable via options I think

**Pull request merged**